libraries/hotstuff/test/test_pacemaker.cpp

```diff
@@ -70,6 +70,18 @@ namespace eosio::hotstuff {
       }
    }
 
+   void test_pacemaker::duplicate(hotstuff_message_index msg_type) {
+      std::vector<test_pacemaker::hotstuff_message> dup;
+      for (const auto& msg_pair : _pending_message_queue) {
+         const auto& [sender_id, msg] = msg_pair;
+         size_t v_index = msg.index();
+         dup.push_back(msg_pair);
+         if (v_index == msg_type)
+            dup.push_back(msg_pair);
+      }
+      _pending_message_queue = dup;
```

Comment:
```suggestion
      _pending_message_queue = std::move(dup);
```

Reply:
Does it matter where the duplicates are added? Should they be together in the vector?

Reply:
Yes, they need to be together to trigger duplicate vote processing. If the message queue is `abcdefghijklmnopqrstuabcdefghijklmnopqrstu` votes, after processing the first `o` vote (15th vote) a quorum is met and `process_vote` correctly won't try to aggregate any more votes. But if you flush the duplicates as  `aabbccddeeffgghhiijjkkllmmnnooppqqrrssttuu`, then the second time `process_vote` sees `a`, it will try to aggregate it again (quorum not met yet) and then a problem occurs.